packages/app-accounts/src/UploadButton.tsx

```diff
@@ -0,0 +1,259 @@
+    const fileReader: FileReader = new FileReader();
+
+    fileReader.onload = (e) => {
+      try {
+        if (!isUndefined(e) && e.target !== null) {
+          const fileContents: any = JSON.parse(e.target.result);
+
+          if (Object.keys(fileContents).includes('address' && 'encoding' && 'meta')) {
+            const json: KeyringPair$Json | undefined = fileContents;
+
+            keyring.addFromJson(json as KeyringPair$Json);
+
+            // Store uploaded wallet in state and open modal to get their password for it
+            this.setState({ uploadedFileKeyringPair: json }, () => this.showPasswordModal());
+          }
+        }
+      } catch (e) {
+        console.error('Error uploading file: ', e);
+        this.setState({ error: { key: t('error'), value: t('Unable to upload account from file') } });
+      }
+    };
+
+    fileReader.readAsText(fileToUpload);
+  }
+
+  processUploadedFileStorage = (): void => {
+    const { password, uploadedFileKeyringPair } = this.state;
+    const { t, onChangeAccount } = this.props;
+
+    if (isUndefined(uploadedFileKeyringPair) || !uploadedFileKeyringPair.address) {
+      return;
+    }
+
+    const json = uploadedFileKeyringPair;
+
+    // Reset password so it is not pre-populated on the form on subsequent uploads
+    this.setState({ password: '' }, () => {
+      try {
+        if (!json || !Object.keys(json).length) {
+          return;
+        }
+
+        const pairRestored = keyring.restoreAccount(json, password);
+
+        if (pairRestored) {
+          this.hidePasswordModal();
+          onChangeAccount(pairRestored.publicKey());
+
+          return;
+        } else {
+          this.setState({ error: { key: t('error'), value: t('Unable to upload account into memory') } });
+
+          return;
+        }
+      } catch (e) {
+        console.error('Error processing uploaded file to local storage: ', e);
+        this.setState({ error: { key: t('error'), value: t('Unable to upload account into memory') } });
+
+        return;
+      }
+    });
+
+    return;
+  }
+
+  showPasswordModal = (): void => {
+    this.setState({ isPasswordModalOpen: true });
+  }
+
+  hidePasswordModal = (): void => {
+    this.setState({ isPasswordModalOpen: false });
+  }
+
+  render () {
+    const { uploadedFileKeyringPair, isPasswordModalOpen } = this.state;
+    const { className, icon = 'upload', isCircular = true, isPrimary = true, size = 'tiny', style } = this.props;
+    const address = uploadedFileKeyringPair && uploadedFileKeyringPair.address;
+
+    return (
+      <div className={'accounts--Address-upload'}>
+        { address && isPasswordModalOpen ? (
+            <Modal
+              dimmer='inverted'
+              open={isPasswordModalOpen}
+              onClose={this.hidePasswordModal}
+              size={'mini'}
+            >
+              <Modal.Content>
+                <div className='ui--grid'>
+                  <div className={'accounts--Address-modal'}>
```

Comment:
Same `className=` comments as per the previous file. Change form this line downwards ...